agent/iskakalunan.py

```diff
@@ -0,0 +1,350 @@
+                if not 'moztpeqa' in message_temp.sender:
+                    logger.debug("message sender: %s" % message_temp.sender)
+                    messages.append(message_temp)
+
+            return messages
+        except errors.HttpError, error:
+            logger.error("An error occurred: %s" % error)
+
+    def prepare_try_build(self, revision):
+        url_try_builds = "https://archive.mozilla.org/pub/firefox/try-builds/"
+        req = urllib2.Request(url_try_builds)
+        response = urllib2.urlopen(req)
+        html_try_builds = response.read()
+
+        pattern = 'href="(.+' + message.revision + '\/)"'
+        m = re.search(pattern, html_try_builds)
+        url_revision_build = "https://archive.mozilla.org" + m.group(1) + "try-win64/firefox-55.0a1.en-US.win64.zip"
+        logger.debug("Download build from %s" % url_revision_build)
+        file_revision_build = urllib2.urlopen(url_revision_build)
+        with open('firefox.zip', 'wb') as output:
+            while True:
+                data = file_revision_build.read(4096)
+                if data:
+                    output.write(data)
+                else:
+                    break
+
+        with zipfile.ZipFile('firefox.zip', 'r') as zf:
+            zf.extractall()
+        shutil.rmtree(r'C:/Program Files/Mozilla Firefox')
+        shutil.move('firefox', r'C:/Program Files/Mozilla Firefox')
+
+    def prepare_job(self, message):
+        """Prepare a json represent a job and update message status
+        """
+        self.prepare_try_build(message.revision)
+
+        cmd = [sys.executable,
+               "runtest.py",
+               "re",
+               "test.suite",
+               "--max-run="+str(message.max_run),
+               "--max-retry="+str(message.max_retry)]
+        if message.calc_si != "":
+            cmd.append(message.calc_si)
+        if message.advance != "":
+            cmd.append(message.advance)
+
+        logger.debug("cmd: %s" % ' '.join(cmd))
+        subprocess.call(cmd, env=os.environ.copy())
+
+    def check_job_status(self):
+        """Check if a job is finished
+
+        return - True if a job is finished
+        """
+        if os.path.isfile(os.path.join(UPLOAD_FOLDER, max(os.listdir(UPLOAD_FOLDER)), RESULT_JSON)):
+            if os.path.isfile(RESULT_ZIP):
+                os.remove(RESULT_ZIP)
+            shutil.make_archive("result", 'zip', os.path.join(UPLOAD_FOLDER, max(os.listdir(UPLOAD_FOLDER))))
+            logger.info("Job finished by checking")
+            return True
+        else:
+            logger.info("Job did not finish by checking")
+            return False
+
+    def reply_status(self, msg_src, status):
+        """Update status to sender"""
+        logger.debug("reply status to %s with %s" % (msg_src.message_id, status))
+        message = MIMEMultipart()
+        message['From'] = 'moztpeqa@mozilla.com'
```

Comment:
Should we consider to put email address as a global variable?  The email address modification could happen in nearly future